tmc-langs-qmake/src/test/java/fi/helsinki/cs/tmc/langs/qmake/QmakePluginTest.java

```diff
@@ -0,0 +1,115 @@
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class QmakePluginTest {
+
+    private final QmakePlugin qmakePlugin;
+
+    public QmakePluginTest() {
+        qmakePlugin = new QmakePlugin();
+    }
+
+    @Test
+    public void testGetLanguageName() {
+        assertEquals("qmake", qmakePlugin.getLanguageName());
+    }
+
+    @Test
+    public void testQtBuildFailingWithNoLib() {
+        assertEquals(RunResult.Status.PASSED, runTests("passing_nolib"));
+    }
+
+    @Test
+    public void testQtBuildFailingWithSingleLib() {
+        assertEquals("Failing compile with single lib",
+                RunResult.Status.COMPILE_FAILED,
+                runTests("failing_compile_single_lib_compiling"));
+    }
+
+    @Test
+    public void testQtLibBuildFailing() {
+        assertEquals(RunResult.Status.COMPILE_FAILED, runTests("failing_single_lib_not_compiling"));
+    }
+
+    @Test
+    public void testQTestsFailingNoLib() {
+        assertEquals(RunResult.Status.TESTS_FAILED, runTests("failing_nolib"));
+    }
+
+    @Test
+    public void testQtTestsPassingSingleLib() {
+        assertEquals(RunResult.Status.PASSED, runTests("passing_single_lib"));
+    }
+
+    @Test
+    public void testQtTestsFailingNoLibSamePoints() {
+        assertEquals("Failing no library",
+                RunResult.Status.TESTS_FAILED,
+                runTests("failing_nolib_same_point"));
+    }
+
+    @Test
+    public void testScanExerciseWithFailingSamePoints() {
+        Optional<ExerciseDesc> optional = scanExercise("failing_single_lib_same_point");
+        assertTrue(optional.isPresent());
+
+        ExerciseDesc desc = optional.get();
+
+        assertEquals("failing_single_lib_same_point", desc.name);
+
+        assertEquals(3, desc.tests.size());
+        assertEquals("test_function_two_here", desc.tests.get(0).name);
+
+        assertEquals(1, desc.tests.get(0).points.size());
+        assertEquals("2", desc.tests.get(0).points.get(0));
+    }
+
+    @Test
+    public void testIsExerciseTypeCorrect() {
+        assertTrue(isExerciseTypeCorrect("passing_nolib"));
+        assertTrue(isExerciseTypeCorrect("passing_nolib_same_point"));
+        assertTrue(isExerciseTypeCorrect("passing_single_lib"));
+        assertTrue(isExerciseTypeCorrect("passing_single_lib_same_point"));
+
+        assertTrue(isExerciseTypeCorrect("passing_multiple_lib"));
+        assertTrue(isExerciseTypeCorrect("passing_multiple_lib_same_point"));
+
+        assertTrue(isExerciseTypeCorrect("failing_single_lib_not_compiling"));
+        assertTrue(isExerciseTypeCorrect("failing_nolib"));
+        assertTrue(isExerciseTypeCorrect("failing_nolib_same_point"));
+        assertTrue(isExerciseTypeCorrect("failing_single_lib"));
+        assertTrue(isExerciseTypeCorrect("failing_single_lib_not_compiling"));
+        assertTrue(isExerciseTypeCorrect("failing_single_lib_same_point"));
+        assertFalse(isExerciseTypeCorrect(""));
```

Comment:
Should this also check against a project that is otherwise valid but just not a QT project?